packages/server/api/webscrape.py

```diff
@@ -3,32 +3,63 @@
 import re
 from urllib import parse
 from bs4 import BeautifulSoup
+import json
 
 class handler(BaseHTTPRequestHandler):
   whitelisted_url_regexes = [
     re.compile('^https:\/\/www\.bonappetit\.com')
   ]
 
+  def error_handler(error_string, error_code = 500):
+    self.send_response(error_code)
+    self.send_header('Content-type', 'text/plain')
+    self.end_headers()
+    self.wfile.write(errror_string.encode())
+
   def do_GET(self):
     url = parse.parse_qs(parse.urlsplit(self.path).query)['url'][0]
     url_in_whitelist = any(regex.match(url) for regex in self.whitelisted_url_regexes)
     
     if(not url_in_whitelist):
-      self.send_response(500)
-      self.send_header('Content-type', 'text/plain')
-      self.end_headers()
-      self.wfile.write('Unsupported recipe site'.encode())
+      error_handler('URL not supported')
+      return
+    
+    recipe_object = {
```

Comment:
I don't love that we define this dictionary with a bunch of empty fields. Maybe we define just a dictionary and add the fields as necessary?

Reply:
Yeah sure